crates/sui-indexer/src/store/module_resolver_v2.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::schema_v2::packages;
+use diesel::ExpressionMethods;
+use diesel::QueryDsl;
+use diesel::RunQueryDsl;
+use move_core_types::language_storage::ModuleId;
+use move_core_types::resolver::ModuleResolver;
+use sui_types::base_types::ObjectID;
+use sui_types::move_package::MovePackage;
+
+use crate::errors::{Context, IndexerError};
+use crate::models_v2::packages::StoredPackage;
+use crate::store::diesel_marco::read_only_blocking;
+use crate::PgConnectionPool;
+
+/// A package resolver that reads packages from the database.
+pub struct IndexerStoreModuleResolver {
+    cp: PgConnectionPool,
+}
+
+impl IndexerStoreModuleResolver {
+    // TODO remove this after integration is done
+    #[allow(dead_code)]
+    pub fn new(cp: PgConnectionPool) -> Self {
+        Self { cp }
+    }
+}
+
+impl ModuleResolver for IndexerStoreModuleResolver {
+    type Error = IndexerError;
+
+    fn get_module(&self, id: &ModuleId) -> Result<Option<Vec<u8>>, Self::Error> {
+        let package_id = ObjectID::from(*id.address()).to_vec();
+        let module_name = id.name().to_string();
+
+        // Note: this implementation is potentially vulnerable to package upgrade race conditions
+        // for framework packages because they reuse the same package IDs.
```

Comment:
Given this comment, is there another better impl?

Reply:
(Coming later)

Reply:
> (Coming later)

Later as in down the PR stack? Or as in later in future improvement?
Whats the drawback & blast radius for letting this go in now?

Reply:
I haven't dealt with this (package upgrade) in my stack. But it's a must-fix before we say indexer-v2 is done.